File: app/src/main/java/com/devserrano/dragonballexplorer/database/DatabaseProvider.java
package com.devserrano.dragonballexplorer.database;

import android.content.Context;

import androidx.room.Room;

public class DatabaseProvider {

    private static AppDatabase database;

    public static AppDatabase getDatabase(Context context) {

        if (database == null) {
            database = Room.databaseBuilder(
                            context.getApplicationContext(),
                            AppDatabase.class,
                            "dragon_ball_database"
                    )
                    .fallbackToDestructiveMigration()
                    .build();
        }

        return database;
    }
}

File: app/src/main/java/com/devserrano/dragonballexplorer/database/DatabaseProvider.kt
package com.devserrano.dragonballexplorer.database

import android.content.Context
import androidx.room.Room

object DatabaseProvider {

    private var database: AppDatabase? = null

    fun getDatabase(context: Context): AppDatabase {

        if (database == null) {

            database = Room.databaseBuilder(
                context.applicationContext,
                AppDatabase::class.java,
                "dragon_ball_database"
            ).build()

        }

        return database!!
    }
}